aws/data_source_aws_iam_roles.go

```diff
@@ -0,0 +1,118 @@
+package aws
+
+import ( // nosemgrep: aws-sdk-go-multiple-service-imports
+	"fmt"
+	"log"
+	"path/filepath" // filepath for glob matching
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAwsIAMRoles() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsIAMRolesRead,
+
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
```

Comment:
the `filters` field is generally used when the AWS API method supports it in a request, so in this case because `ListRoles` doesn't, we can instead use an argument like `name_prefix` or `name_regex` to let users specify that input at the root of the resource. `data_source_aws_ami.go` is a good example of the `name_regex` usage.